### PR TITLE
docs: document VS Code extension

### PR DESCRIPTION
**Pull request by dtarnawsky**
This adds a new expandable section in the docs for the VS Code Extension:
1. Getting Started
2. New Project
3. Web to Native
4. Build and Run
5. Debugging
6. Bundle Analysis
7. Migrate from Cordova
8. Dependencies
9. Framework Features
10. Mono Repos
11. Native Settings
12. Recommendations
13. Release to the Store
14. Running Scripts
15. Splash Screen & Icon
16. Advanced Topics

### CODE REVIEW

File: docs/main/vs-code-ext/0-getting-started.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+<h2>Next Steps</h2>
+
+- [Create a new project](new-project)
+- [Make an existing web app native](web-to-native)
+- [Migrating from Cordova to Capacitor](cordova)
+
+Also learn the essentials of [building and running](build-and-run), [debugging](debugging) and [preparing](release) your app for release with a [Splash Screen & Icon](splash-icon), and setting [version information](native-settings).
+
+Finally, improve your project by [maintaining dependencies](dependencies), [minimizing javascript](bundle-cost) usage, and following [recommendations](recommendations).
```
Comment (jameson-w-taylor) [inline docs/main/vs-code-ext/0-getting-started.md:24]: This link is broken (possibly should just be "bundle"?)

File: docs/main/vs-code-ext/1-new-project.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+---
+
+Follow these steps to create a **new starter project** for Angular, React or Vue. 
+
+<video style={{borderRadius:5}} width="100%" controls>
+      <source src={require('@site/static/video/new-project.mp4').default} type="video/mp4" />
+</video>
+
+1. Create a folder for your project and open it in VS Code
+2. Click the `Ionic` logo on the side menu
```
Comment (jameson-w-taylor) [inline docs/main/vs-code-ext/1-new-project.md:16]: Maybe consider standardizing on how to refer to the extension icon in the side menu?

`Web to Native` refers to it as:
>Click the Ionic extension on the side menu

while this page calls it a logo.

File: docs/main/vs-code-ext/3-build-and-run.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+description: Visual Studio Code Extension for Capacitor
+contributors:
+  - dtarnawsky
+slug: /vscode/build-and-run
+---
+
+There are a few different ways to build or run your app depending on your preference and whether you want to test in a web browser or a real or emulated mobile device.
+
+## Testing for Web
+Click `Run` > `Web` to build and launch the options windows:
```
Comment (jameson-w-taylor) [inline docs/main/vs-code-ext/3-build-and-run.md:12]: Should windows be plural?

File: docs/main/vs-code-ext/3-build-and-run.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+The **Live Reload** feature allows you to run your app on a mobile device and whenever a code change is made in VS Code and saved it will reload the application.
+
+To turn on this feature click `Settings` > `Live Reload`
+
+## Using HTTPS
+
+The feature (`Settings` > `Use HTTPS`) will create a certificate and serve your application using HTTPS. Instructions to trust the certificate on web, iOS and Android are displayed.
+
+The option to serve your application via `HTTPS` is requires if you make Web API calls that require a [secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) like [Geolocation](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/geolocation).
```
Comment (jameson-w-taylor) [inline docs/main/vs-code-ext/3-build-and-run.md:45]: "via `HTTPS` is requires", should say "required"

File: docs/main/vs-code-ext/4-debugging.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+slug: /vscode/debugging
+---
+
+Debugging your application using either [the extension](#debug-in-vs-code) or [attaching to the web view](#attach-to-web-view). You can alternatively use [Remote Logging](#remote-logging) or the [Capacitor Preview](#capacitor-preview) app to perform `console.log` style debugging.
+
+## Debug in VS Code
+
+Click the `Debug` item to launch a web browser or attach to a running Android web view for debugging.
+
+Click `Debug` > `Web` to launch a _debuggable_ web browser such as Chrome or MS Edge. This will build your app then put VS Code into debugging mode allowing you set breakpoints, inspect variables etc.
```
Comment (jameson-w-taylor) [inline docs/main/vs-code-ext/4-debugging.md:15]: "... allowing you set breakpoints ...", missing the word "to"

File: docs/main/vs-code-ext/cordova.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+:::
+
+### Unneeded Plugins
+Cordova plugins that you **no longer need** will be flagged and you can click `Uninstall` to remove them.
+
+### Incompatible Plugins
+Cordova plugins that are on a known **[incompatible list](https://capacitorjs.com/docs/plugins/cordova#known-incompatible-plugins)** will be flagged. There may be Capacitor equivalents, your code will need to be refactored.
+
+### Better Plugins
+Cordova plugins that have **better** equivalent Capacitor plugins will show as optional suggestions (light bulb icon). A **better** plugin is defined as one that has official support from the Capacitor team. Some Cordova plugins have been deprecated or are no longer maintained, we track these a provide suggestions for alternatives.
```
Comment (jameson-w-taylor) [inline docs/main/vs-code-ext/cordova.md:28]: "... we track these a provide ...", should say "and"

File: docs/main/vs-code-ext/dependencies.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+Click `X packages` to show all dependencies showing their current version and their latest version.
+
+![List of dependencies](/img/dependency.png)
+
+Click a Dependency and Choose:
+- `Upgrade` to upgrade the dependency to the latest version
+- `Info` to display the npm web page for this dependency
+- `Uninstall` to remove the dependency from the project
+
+Some dependencies that require additional migration steps may be done automatically for you. For example: updating `@ionic/angular-toolkit` will migration `angular.json` to remove unneeded sections or add `@ionic/cordova-builders` for Cordova projects.
```
Comment (jameson-w-taylor) [inline docs/main/vs-code-ext/dependencies.md:23]: "... will migration ...", should be "migrate"

**Review by dallastjames — APPROVED** [review]
Read through and it looks great! Exciting to get these docs added! 🚀

**Pull request merged**